services/time/examples/simple_service.rs

```diff
@@ -71,28 +73,35 @@ impl<'a> MarkerSchema<&'a mut Fork> {
 
 transactions! {
     MarkerTransactions {
-        const SERVICE_ID = SERVICE_ID;
 
         /// Transaction, which must be executed no later than the specified time (field `time`).
         struct TxMarker {
-            from: &PublicKey,
             mark: i32,
             time: DateTime<Utc>,
         }
     }
 }
 
-impl Transaction for TxMarker {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.from())
+impl TxMarker {
+    fn sign(
+        mark: i32,
+        time: DateTime<Utc>,
+        pk: &PublicKey,
+        sc: &SecretKey,
+    ) -> Message<RawTransaction> {
+        Protocol::sign_transaction(TxMarker::new(mark, time), SERVICE_ID, *pk, sc)
```

Comment:
If we have a vertical argument list then we can as well use proper variable names like `public_key` instead of `pk`. Though, it's a one-line function so it may be okay to keep them short. Just be consistent with **k**'s and **c**'s, please :)

On a side note, why public key needs to be dereferenced but secret key doesn't? That's also a bit inconsistent.

(Ditto for `TxTime`.)

Reply:
This method used for backward compatibility with old signature. Thats why Pk is dereferenced (in new messages it's used by value).